oak_functions/loader/src/lookup.rs

```diff
@@ -14,247 +14,140 @@
-    /// method, and existing entries are left untouched. The caller may retry the refresh operation
-    /// at a future time.
-    pub async fn refresh(&self) -> anyhow::Result<()> {
-        match &self.lookup_data_source {
-            Some(lookup_data_source) => {
-                let start = Instant::now();
-                let lookup_data_buf = fetch_lookup_data(&self.logger, lookup_data_source).await?;
-                self.logger.log_public(
-                    Level::Info,
-                    &format!(
-                        "fetched {} bytes of lookup data in {:.0?}",
-                        lookup_data_buf.len(),
-                        start.elapsed()
-                    ),
-                );
-
-                let start = Instant::now();
-                let entries = parse_lookup_entries(&mut lookup_data_buf.as_ref())
-                    .context("could not parse lookup data")?;
-
-                self.logger.log_public(
-                    Level::Info,
-                    &format!(
-                        "parsed {} entries of lookup data in {:.0?}",
-                        entries.len(),
-                        start.elapsed()
-                    ),
-                );
-
-                // This block is here to emphasize and ensure that the write lock is only held for a
-                // very short time.
-                let start = Instant::now();
-                {
-                    *self
-                        .entries
-                        .write()
-                        .expect("could not lock entries for write") = entries;
-                }
-                self.logger.log_public(
-                    Level::Debug,
-                    &format!(
-                        "lookup data write lock acquisition time: {:.0?}",
-                        start.elapsed()
-                    ),
-                );
-
-                Ok(())
-            }
-            None => Ok(()),
-        }
-    }
-
-    /// Creates an instance of LookupData populated with the given entries.
-    #[allow(dead_code)]
-    pub fn for_test(entries: HashMap<Vec<u8>, Vec<u8>>) -> Self {
-        LookupData {
-            lookup_data_source: None,
-            entries: RwLock::new(entries),
-            logger: Logger::for_test(),
-        }
-    }
-
-    /// Convenience getter for an individual entry that reduces lock contention by cloning the
-    /// resulting value as quickly as possible and returning it instead of a reference.
-    pub fn get(&self, key: &[u8]) -> Option<Vec<u8>> {
-        self.entries
-            .read()
-            .expect("could not lock entries for read")
-            .get(key)
-            .cloned()
-    }
-
-    #[allow(dead_code)]
-    pub fn len(&self) -> usize {
-        self.entries
-            .read()
-            .expect("could not lock entries for read")
-            .len()
+    /// Create and return a lookup factory.
+    pub async fn create(
```

Comment:
This function does not need to be async. Does it?

Reply:
You are right, it does not. 

I made it async because I wanted it to have the same signature as the corresponding functions in tf / metric, and those need to be async. 

Do you think I should change it?

Reply:
Are you referring to `create_tensorflow_factory` and `create_metrics_proxy_factory`? 

I think in general we should avoid unnecessary asyncs. Because that complicates the code as it results in unnecessary awaits and blocking calls on tokio runtime.

Also I think `create_metrics_proxy_factory` does not have to be async. Perhaps fix that too?

Reply:
Yes, I meant `create_tensorflow_factory` and `create_metrics_proxy_factory`. 

Perfect, so I will remove the async from this `Lookup::create` and in a seperate PR remove the async from `create_metrics_proxy_factory` too.

SG?

Reply:
Yes. I agree that `create_metrics_proxy_factory` can be fixed in a separate PR.